### ISSUE TITLE
DetectDelimiter() throws exception for some special delimiters

### ISSUE THREAD

**Issue by MacDisein** [JoshClose/CsvHelper#1786]
We use the character "þ" as delimiter and this throws a system exception "unrecognized escape sequence".

https://github.com/JoshClose/CsvHelper/blob/1d37c0b7a2ce83a82db7e87388abf0d92927f88d/src/CsvHelper/CsvParser.cs#L266

Escaping a character is not always correct, so not every character should be escaped - only the characters with special meaning.
Refer: https://docs.microsoft.com/en-us/dotnet/standard/base-types/character-escapes-in-regular-expressions

You may use the following code to avoid exceptions:
```c#
string[] specialMeaningCharacters = new[] { ".", "$", "^", "{", "[", "(", "|", ")", "*", "+", "?", "\\" };
var delimiterCounts = new Dictionary<string, int>();
foreach (var delimiter in delimiterValues)
{
      var pattern = delimiter;
      if(specialMeaningCharacters.Any(delimiter.Contains))
      {
            // Escape every special meaning character to use as regex pattern.
            pattern = Regex.Replace(delimiter, "(.)", "\\$1");
       }
       delimiterCounts[delimiter] = Regex.Matches(text, pattern).Count;
}
```

So only the characters with special meaning are being escaped.

**Comment by JoshClose** [comment]
Thanks! The character to look for can just be changed to that set of chars.

```cs
var delimiterCounts = new Dictionary<string, int>();
foreach (var delimiter in delimiterValues)
{
	// Escape regex special chars to use as regex pattern.
	var pattern = Regex.Replace(delimiter, @"([.$^{\[(|)*+?\\])", "\\$1");
	delimiterCounts[delimiter] = Regex.Matches(text, pattern).Count;
}
```

**Comment by JoshClose** [comment]
This fix is in 27.0.2 on NuGet.
